Sources/PackageRegistry/RegistryClient.swift

```diff
@@ -325,24 +330,49 @@ public final class RegistryClient {
         )
 
         self.httpClient.execute(request, observabilityScope: observabilityScope, progress: nil) { result in
-            completion(result.tryMap { response in
-                try self.checkResponseStatusAndHeaders(response, expectedStatusCode: 200, expectedContentType: .json)
+            switch result {
+            case .success(let response):
+                do {
+                    try self.checkResponseStatusAndHeaders(response, expectedStatusCode: 200, expectedContentType: .json)
 
-                guard let data = response.body else {
-                    throw RegistryError.invalidResponse
-                }
+                    guard let data = response.body else {
+                        throw RegistryError.invalidResponse
+                    }
 
-                let versionMetadata = try self.jsonDecoder.decode(Serialization.VersionMetadata.self, from: data)
-                guard let sourceArchive = versionMetadata.resources.first(where: { $0.name == "source-archive" }) else {
-                    throw RegistryError.missingSourceArchive
-                }
+                    let versionMetadata = try self.jsonDecoder.decode(Serialization.VersionMetadata.self, from: data)
+                    guard let sourceArchive = versionMetadata.resources.first(where: { $0.name == "source-archive" }) else {
+                        throw RegistryError.missingSourceArchive
+                    }
 
-                guard let checksum = sourceArchive.checksum else {
-                    throw RegistryError.invalidSourceArchive
-                }
+                    guard let checksum = sourceArchive.checksum else {
+                        throw RegistryError.invalidSourceArchive
+                    }
 
-                return checksum
-            })
+                    if let fingerprintStorage = self.fingerprintStorage {
+                        fingerprintStorage.put(package: package,
+                                               version: version,
+                                               fingerprint: .init(origin: .registry(registry.url), value: checksum),
+                                               observabilityScope: observabilityScope,
+                                               callbackQueue: callbackQueue) { storageResult in
+                            switch storageResult {
+                            case .success:
+                                completion(.success(checksum))
+                            case .failure(let error):
+                                if case PackageFingerprintStorageError.conflict(_, let existing) = error {
```

Comment:
nice 😍

Reply:
`case .failure(PackageFingerprintStorageError.conflict(_, let existing)` may also work instead of a nested if